src/ImageSharp.Web/FormatUtilities.cs

```diff
@@ -50,32 +55,46 @@ public FormatUtilities(IOptions<ImageSharpMiddlewareOptions> options)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public string GetExtensionFromUri(string uri)
         {
-            // TODO: Investigate using span to reduce allocations here.
-            string[] parts = uri.Split('?');
-            if (parts.Length > 1 && QueryHelpers.ParseQuery(parts[1]).TryGetValue(FormatWebProcessor.Format, out StringValues ext))
+            int query = uri.IndexOf('?');
+            ReadOnlySpan<char> path;
+
+            if (query > -1)
             {
-                return ext;
+                if (uri.Contains(FormatWebProcessor.Format, StringComparison.OrdinalIgnoreCase) && QueryHelpers.ParseQuery(uri.Substring(query)).TryGetValue(FormatWebProcessor.Format, out StringValues ext))
+                {
+                    return ext;
+                }
+
+#if !NETCOREAPP3_1_OR_GREATER
+                path = uri.ToLowerInvariant().AsSpan(0, query);
+#else
+                path = uri.AsSpan(0, query);
+#endif
+            }
+            else
+            {
+#if !NETCOREAPP3_1_OR_GREATER
+                path = uri.ToLowerInvariant();
```

Comment:
No need to use `ToLowerInvariant` since the comparison are later done using `OrdinalIgnoreCase`. That would remove this allocation, and maybe make the whole thing alloc-free.

Reply:
Correct, this was a left over from previous version, removed 👍